src/oop_ext/foundation/decorators.py

```diff
@@ -92,35 +94,36 @@ def Wrapper(func):
     return Wrapper
 
 
-def Deprecated(name=None):
+def Deprecated(what: Optional[object] = None) -> Callable:
```

Comment:
Watch out for API breakage :)

Reply:
Heh yeah, but in this case `name` is clearly wrong, because it can be either a string or a function (anything with a `__name__` really), so decided it was best to rename the attribute (I doubt someone uses `@Deprecated(name=...)`, but the fix is simple).